In [58]:
import numpy as np
import scipy.stats as stats

## Задание 1. 
Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.

In [59]:
s_dev = 16
alpha = 0.05
M = 80
n = 256
z_value_half_alpha = 1.96

### conf_interval = M+-Z(a/2)*s_dev/sqrt(n)

In [60]:
conf_interval_plus = M+z_value_half_alpha*s_dev/np.sqrt(n)

In [61]:
conf_interval_plus

81.96

In [62]:
conf_interval_minus = M-z_value_half_alpha*s_dev/np.sqrt(n)

In [63]:
conf_interval_minus

78.04

### Ответ: [78.04,81.96]

## Задание 2
В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.

In [64]:
alpha = 0.05
x_2 = [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]
n_2 = len(x_2)

In [65]:
M_2 = np.mean(x_2)

In [66]:
M_2

6.590000000000001

In [67]:
s_dev_2 = np.std(x_2,ddof = 1)

In [68]:
s_dev_2

0.4508017549014448

In [69]:
conf_inverval_lower = M_2 - z_value_half_alpha*s_dev_2/np.sqrt(n_2)

In [70]:
conf_inverval_lower

6.3105901775368505

In [71]:
conf_inverval_higher = M_2 + z_value_half_alpha*s_dev_2/np.sqrt(n_2)

In [72]:
conf_inverval_higher

6.869409822463151

### Ответ: истинное значение величины X лежит в интервале [6.31, 6.87]

## Задание 3
Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170
Рост матерей 178, 165, 165, 173, 168, 155, 160, 164, 178, 175

Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.


In [73]:
D_hight = [175, 167, 154, 174, 178, 148, 160, 167, 169, 170]

In [74]:
M_hight = [178, 165, 165, 173, 168, 155, 160, 164, 178, 175]

In [75]:
alpha_3 = 0.05

In [76]:
len(D_hight)

10

In [77]:
len(M_hight)

10

In [78]:
D_mean = np.mean(D_hight) # средний рост среди дочерей

In [79]:
M_mean = np.mean(M_hight) # средний рост среди матерей

In [80]:
delta = D_mean - M_mean
delta

-1.9000000000000057

In [81]:
D_var = np.var(D_hight,ddof = 1)
D_var

91.06666666666666

In [82]:
M_var = np.var(M_hight,ddof = 1)
M_var

60.10000000000001

In [83]:
Var_combined = (D_var+M_var)/2 # объединенная дисперсия
Var_combined

75.58333333333334

In [84]:
SE = np.sqrt(Var_combined/10 + Var_combined/10)
SE

3.8880157750022915

In [86]:
t = stats.t.ppf(0.975,18)
t

2.10092204024096

In [87]:
conf_lower = delta - t*SE
conf_lower

-10.068418034506857

In [88]:
conf_higher = delta + t*SE
conf_higher

6.268418034506846

### Ответ: доверительный интервал для разности роста дочерей и матерей [-10.06, 6.27]